### PR TITLE
Add org.munadi.Munadi

### PR DESCRIPTION
**Pull request by hakpw**
Initial release

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.munadi.Munadi

**Comment by hakpw** [comment]
@nedrichards I just checked https://flathub.org/builds/#/ it says it failed!
How is that possible?
I did multiple tests here locally on Fedora 28 and made sure it is fully functional.

File: org.munadi.Munadi.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+    "app-id": "org.munadi.Munadi",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
+    "sdk": "org.kde.Sdk",
+    "command": "munadi",
+    "branch": "17.02",
```
Comment (nedrichards) [inline org.munadi.Munadi.json]: Remove this. For flathub this *has* to be 'stable' which is set automatically.

**Comment by nedrichards** [comment]
@hakpw it's a test that checks for the flatpak branch being "stable" https://flathub.org/builds/#/builders/32/builds/551 so just remove the line I've highlighted in your manifest and everything will be OK on a rebuild.

**Comment by hakpw** [comment]
@nedrichards Thanks for the tip.

The part talking about that on the doc was vague, I remember getting confused as to what it meant, now I understand :)

**Comment by nedrichards** [comment]
Thanks for the feedback! I'll try and remember to take a look and see if I can make it clearer to people who are less knee deep in flatpak internals.

**Comment by hakpw** [comment]
@nedrichards found it:
"Branches and runtimes" @ https://github.com/flathub/flathub/wiki/App-Requirements

To be honest, I don't know what to suggest instead, if others understood it then you might as well leave it as it is!

Thank you for all this work, I hope this will be the end of GNU/Linux app deployment nightmare :)

File: org.munadi.Munadi.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "munadi",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc", 
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=network",
+        "--socket=pulseaudio",
+        "--filesystem=xdg-config/autostart:create",
```
Comment (AsciiWolf) [inline org.munadi.Munadi.json]: I am not sure if allowing the app to create an autostart file is a good idea as it should bypass sandbox this way. Does the app know how to handle this so it won't start without the sandbox when running in Flatpak?

File: org.munadi.Munadi.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            "name": "munadi",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 org.munadi.Munadi.desktop /app/share/applications/org.munadi.Munadi.desktop",
+                "install -Dm644 org.munadi.Munadi.appdata.xml /app/share/appdata/org.munadi.Munadi.appdata.xml",
+                "tar xf org.munadi.Munadi.icons.tar.gz -C /app/share/",
+                "mv src/* .",
+                "qmake munadi.pro",
+                "make -j4",
```
Comment (AsciiWolf) [inline org.munadi.Munadi.json]: It's better to use `-j $FLATPAK_BUILDER_N_JOBS` instead of `-j4`.

File: org.munadi.Munadi.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.munadi.Munadi.appdata.xml]: Submit this file upstream please.

File: org.munadi.Munadi.json (hunk 2 above)
Comment (barthalion) [reply org.munadi.Munadi.json]: Yes, it has to go.

File: org.munadi.Munadi.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "modules": [
+        {
+            "name": "munadi",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 org.munadi.Munadi.desktop /app/share/applications/org.munadi.Munadi.desktop",
+                "install -Dm644 org.munadi.Munadi.appdata.xml /app/share/appdata/org.munadi.Munadi.appdata.xml",
+                "tar xf org.munadi.Munadi.icons.tar.gz -C /app/share/",
+                "mv src/* .",
```
Comment (barthalion) [inline org.munadi.Munadi.json]: ???

File: org.munadi.Munadi.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                "qmake munadi.pro",
```
Comment (barthalion) [inline org.munadi.Munadi.json]: Use `qmake` buildsystem instead.

File: org.munadi.Munadi.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                    "type": "file",
+                    "path": "org.munadi.Munadi.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "org.munadi.Munadi.appdata.xml"
+                },
+                {
+                    "type": "file",
+                    "path": "org.munadi.Munadi.icons.tar.gz"
```
Comment (barthalion) [inline org.munadi.Munadi.json]: What is this for? If these icons aren't part of upstream repo, where did you get them from?

**Comment by hadess** [comment]
FWIW, I couldn't get "munadi.org" to answer, so the reverse-domain name is probably wrong. Should probably be `com.gitlab.munadi.Munadi` according to the URL used in the AppData.

File: org.munadi.Munadi.json (hunk 2 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: @AsciiWolf well spotted, that's the safest way I found.

Comment (hakpw) [reply org.munadi.Munadi.json]: At least until we get an appropriate portal

File: org.munadi.Munadi.json (hunk 3 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: Thanks, will do.

File: org.munadi.Munadi.appdata.xml (hunk 4 above)
Comment (hakpw) [reply org.munadi.Munadi.appdata.xml]: Will do thanks. And yes, I am the author of the app.

File: org.munadi.Munadi.json (hunk 5 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: I couldn't cd into src then invoke qmake, source code is under src dir, how can I do it?
I could do qmake src/munadi.pro if that's better?

File: org.munadi.Munadi.json (hunk 6 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: Not mentioned on the doc:
http://docs.flatpak.org/en/latest/manifests.html#supported-build-systems

Where can I find info about it? 
Also, how can I tell it to look for the .pro file under src dir?
Thanks.

File: org.munadi.Munadi.json (hunk 7 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: they could be under upstream, does it matter?
I generated them from 480x480 PNG source, author of app here.

**Comment by hakpw** [comment]
@hadess I own the domain, upgrading to Ubuntu 18.04 broke LXD and now restoring everything one bit at a time, Docker this time :-)

Does it matter? If so, I'll priorities the restore.

**Comment by hadess** [comment]
> @hadess I own the domain, upgrading to Ubuntu 18.04 broke LXD and now restoring everything one bit at a time, Docker this time :-)
> 
> Does it matter? If so, I'll priorities the restore.

I think that it should at least match the website link in the appdata file?

File: org.munadi.Munadi.json (hunk 5 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: Found a better way [1], will update.
1. http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html#flatpak-manifest

File: org.munadi.Munadi.json (hunk 6 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: If I use qmake, do I need a new flatpak model to handle icons, appdata ... etc?
Both data and src are part of the same repo.
Or do I reference the same repo twice in two different modules?

File: org.munadi.Munadi.json (hunk 2 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: @barthalion is my solution acceptable? Sandbox is still intact also ability to run at startup is critical for this app!
Thanks.

**Comment by nedrichards** [comment]
@hakpw I'd really like it if the app installed the icons, desktop file and appdata file with the qmake build upstream and then just use the qmake buildsystem in the flatpak manifest to install releases from upstream without mantaining these external files in both places.

I know that you *are* upstream, but I know that this'll be a much better way of managing things.

**Comment by hakpw** [comment]
bot, build org.munadi.Munadi

File: org.munadi.Munadi.json (hunk 2 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: Done

File: org.munadi.Munadi.json (hunk 5 above)
Comment (hakpw) [reply org.munadi.Munadi.json]: Done

**Comment by nedrichards** [comment]
bot, build org.munadi.Munadi

File: org.munadi.Munadi.json (hunk 2 above)
Comment (barthalion) [reply org.munadi.Munadi.json]: Sorry, I was away. I disagree about reasoning. Users can add it to autostart manually in their DE themselves.

Comment (hakpw) [reply org.munadi.Munadi.json]: @barthalion 
I'm sorry, I disagree with your disagreement.

Expecting the user to manually add it is horrible idea from UX prospective.
There is no built-in way to do it in gnome even! So limiting the workaround is not acceptable unless a better sane solution exists.

Thanks.

**Comment by nedrichards** [comment]
Fair enough.

**Pull request closed without merging**